### PR TITLE
Feature/service logic ports adapters

### PR DESCRIPTION
**Pull request by devtomasjr**
Add product ports, adapters and service

### CODE REVIEW

File: src/Tests/Application/Services/ProductService.test.ts [hunk 1]
```diff
@@ -0,0 +1,75 @@
+import ProductService from '@Application/services/ProductService'
+import InMemomryProductRepository from '@Adapters/Secondary/DataInMemory/Repositories/ProductRepositoryInMemory'
```
Comment (evilfeeh) [inline src/Tests/Application/Services/ProductService.test.ts:2]: **comment** (for fun): someone mispelling 'memory' hahah

File: src/Adapters/Secondary/DataInMemory/Repositories/ProductRepositoryInMemory.ts [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        return Right<string>("Product removed")
+    }
+
+    async findByCategory(category: string): Promise<Either<Error, Product[]>> {
+        let products = undefined
+        const productSaved = this.list.filter(
+            (productFind) => productFind.category === category
+        )
+
+        if (productSaved) {
```
Comment (evilfeeh) [inline src/Adapters/Secondary/DataInMemory/Repositories/ProductRepositoryInMemory.ts:75]: **suggestion** (non-blocking): early return

We can improve this part doing the early return, this way we can be more objective and avoid callback hell

File: src/Adapters/Secondary/MySqlAdapter/Repositories/ProductRepository.ts [hunk 3]
```diff
@@ -0,0 +1,109 @@
+import { Repository } from 'typeorm';
+import Product from '@Entities/Product'
+import IProductRepository from '@Application/Ports/Secondary/IProductRepository'
+import { Either, Left, Right } from '@Shared/util/either'
```
Comment (evilfeeh) [inline src/Adapters/Secondary/MySqlAdapter/Repositories/ProductRepository.ts:4]: **issue**(blocking): module path

File: src/Adapters/Secondary/MySqlAdapter/Repositories/ProductRepository.ts [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+        id,
+      })
+
+      if (!productToRemove) {
+        return Left<Error>(new NotFoundException('Product: ' + id))
+      }
+
+      await this.repository.remove(productToRemove);
+
+      return Right<string>('Product has been removed')
```
Comment (evilfeeh) [inline src/Adapters/Secondary/MySqlAdapter/Repositories/ProductRepository.ts:66]: Question (non-blocking): Try catch use

Why your not using trycatch in this block? is there some specific reason or it's just useless?

**Review by evilfeeh — COMMENTED** [review]
Almost there, just need to fix the imports

**Review by marceloribeirosilva — APPROVED** [review]

**Review by barreto0 — APPROVED** [review]

File: src/Adapters/Secondary/MySqlAdapter/Repositories/ProductRepository.ts (hunk 4 above)
Comment (devtomasjr) [reply src/Adapters/Secondary/MySqlAdapter/Repositories/ProductRepository.ts:66]: Porque está apenas chamando o método do repository que está inteiramente coberto pelo try catch, então o erro é só repassado pelo either conforme o contrato

**Pull request merged**